### PR TITLE
[wip][generated] source: spec3.sdk.yaml@spec-cb3438d in master

### PR DESCRIPTION
**Pull request by mickjermsurawong-stripe**
r? @remi-stripe 
This is from latest spec. Does it look right to you?

### CODE REVIEW

File: src/main/java/com/stripe/model/CreditNote.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  /** The link to download the PDF of the credit note. */
+  @SerializedName("pdf")
+  String pdf;
+
+  /**
+   * Reason for issuing this credit note, one of `duplicate`, `fraudulent`, `order_cancellation`,
+   * `order_change`, `other`, or `product_unsatisfactory`.
+   */
+  @SerializedName("reason")
+  String reason;
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/CreditNote.java]: This is not correct because it hasn't been updated per the latest API review yet.

File: src/main/java/com/stripe/model/Customer.java [hunk 2]
```diff
@@ -274,6 +274,28 @@ public Customer delete(Map<String, Object> params, RequestOptions options)
...
+  public CustomerBalanceTransactionCollection balanceTransactions(
+      Map<String, Object> params, RequestOptions options) throws StripeException {
+    String url =
+        String.format(
+            "%s%s",
+            Stripe.getApiBase(),
+            String.format("/v1/customers/%s/customer_balance_transactions", this.getId()));
+    return requestCollection(url, params, CustomerBalanceTransactionCollection.class, options);
+  }
+
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/Customer.java]: Flagging that since CreditNote and BalanceTransaction are not going to be released at the same time, we will need to do 2 separate PRs.

File: src/main/java/com/stripe/model/CustomerBalanceTransaction.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
+// Generated by com.stripe.generator.entity.SdkBuilder
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/CustomerBalanceTransaction.java]: same as above, this will need to be a separate PR

File: src/main/java/com/stripe/model/CustomerCollection.java [hunk 4]
```diff
@@ -2,4 +2,28 @@
...
+  }
+
+  /** Retrieves a specific transaction applied to a customer’s balance. */
+  public Customer retrieve(String id, Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    String url =
+        String.format("%s%s", Stripe.getApiBase(), String.format("%s/%s", this.getUrl(), id));
+    return ApiResource.request(ApiResource.RequestMethod.GET, url, params, Customer.class, options);
+  }
+}
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/CustomerCollection.java]: Can you clarify what is happening here? Looks like we put the methods on the wrong collection?

File: src/main/java/com/stripe/model/PaymentIntent.java [hunk 5]
```diff
@@ -74,7 +74,7 @@
   @SerializedName("client_secret")
   String clientSecret;
 
-  /** Confirmation method of this PaymentIntent, one of `secret` or `publishable`. */
+  /** Confirmation method of this PaymentIntent, one of `manual` or `automatic`. */
```
Comment (remi-stripe) [inline src/main/java/com/stripe/model/PaymentIntent.java]: This is kinda secret and should not be in this PR today (not a big deal though)

**Comment by mickjermsurawong-stripe** [comment]
Closing this PR because we will release credit notes and customer balance txn separately.
Addressed the comment on `CustomerCollection` with updating pay-server client metadata on pay-server #140820
Thank you @remi-stripe!
